### PR TITLE
Akhter Company DLC 1 (2?)

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

Its me back again with my bs, so, what's on the table today?

You remember back to the original colony fabricator pr? Where at one point there was a bunch of other stuff, but it got removed at some point? Yeah? No? It doesn't matter either way really because: IT'S FUCKIN BACK BABY!

So, lets get the rundown.

image

**The GPS Beacon**

A semi-stationary beacon that marks a location on GPS, handy for marking where your soon to be new home is!

image

**The Hydroponics Chemical Synthesizer**

The compliment to the water synthesizer, this bad boy makes most of the basic fertilizers and pest control chemicals for piping into hydroponics trays. Get-a-growing!

image

**The Sustenance Dispenser**

Like a soda or booze dispenser, but nowhere near as appetizing. Featuring such amazing dispensing reagents as:
- Astrotame
- Powdered drinks
- Nutraslop
- And more!

image

**The Organic Materials Printer**

Similar to a biogenerator, or the *other*, *lamer* colony core. This'll print a variety of useful goods, such as clothing, equipment, basic emergency medical supplies, plastic, and a small selection of hardy seeds for when you get nothing good to start with!

Now, for some of the new clothing.

image

**Survive the frontier in style!**

Baggy jumpsuits, boots that will never break in, gloves that don't even protect you from electricity fully!

The gas mask is similar to an atmospherics gas mask, with exception of being blazing green.

There is a new radio headset, which while not protecting you from loud sounds at all, will allow you to speak across your z-level as if it was a station bounced radio.

And those two fancy coats there? They're not all that good protection or anything, but they'll hold all sorts of lifesaving gear for you.

image

Protect yourself in style with the new soft helmet and flak jacket, comparable to if not weaker than security equipment.

Save lives with fashion with the new stand-out bright red medical clothing, never be mistaken for a non-medic again!

image

**The Frontier Hazard MOD**

A MOD that can fit in your backpack, and protects you readily from most common external threats! What's not to love! Well... That compressed plates module **cannot** be removed, so you can't use the MOD to store anything.

## How This Contributes To The Skyrat Roleplay Experience

There's two problems this solves for me:

1. There were a few gaps in the capabilities of the colony fabricator, namely in the department of caring for yourself after you've built a little cave in the middle of nowhere. This fills in some of the gaps.

2. The lack of more niche content from the colony fabricator. There's a lot of general stuff, and a lot of engineering stuff, but most people really want it for the tools as they don't have much use for the rest of the stuff. With this I hope to introduce at least a little other use for people.

## Proof of Testing

image

## Changelog

:cl:
add: Several new life and living machines have been added to the rapid construction fabricator, and the company with it. These items include a sustenance dispenser, hydroponics chemical synthesizer, and an all new organic materials fabricator. Check them out today!
add: A whole range of new clothing and equipment from the Akhter company, and from the organic materials fabricator. Colonize in style, now!
qol: Feeling lost? The rapid construction fabricator can now print large, bulky GPS beacons for marking locations.
image: Sprites for all of the new colony fabricator stuff is done by yours truly.
/:cl:

### CODE REVIEW

**Comment by xXPawnStarrXx** [comment]
Could we get stabilizing agent added to the plant chemical creator as a way to lower instability?

**Comment by Paxilmaniac** [comment]
> Could we get stabilizing agent added to the plant chemical creator as a way to lower instability?

How difficult is it, generally, to get that? This is really meant to be just really basic stuff so you can set and forget some botany trays while doing something else.

**Comment by xXPawnStarrXx** [comment]
> How difficult is it, generally, to get that? This is really meant to be just really basic stuff so you can set and forget some botany trays while doing something else.

It's only iron, hydrogen and oxygen. It's fairly easy to get respectively and only benefits one plant stat.

**Comment by Paxilmaniac** [comment]
> > How difficult is it, generally, to get that? This is really meant to be just really basic stuff so you can set and forget some botany trays while doing something else.
> 
> It's only iron, hydrogen and oxygen. It's fairly easy to get respectively and only benefits one plant stat.

Ah. You know this is the same stabilizing agent like for bombs and stuff. I think I'll have to pass on this one unfortunately.

**Comment by xXPawnStarrXx** [comment]
> > > How difficult is it, generally, to get that? This is really meant to be just really basic stuff so you can set and forget some botany trays while doing something else.
> > 
> > 
> > It's only iron, hydrogen and oxygen. It's fairly easy to get respectively and only benefits one plant stat.
> 
> Ah. You know this is the same stabilizing agent like for bombs and stuff. I think I'll have to pass on this one unfortunately.

Yeah, that's fair. It's just inconvenient to lower instability normally. I might try get a merge done on /TG/ reworking the base chems for botany so they are useful all shift.

**Comment by W4rd3nn** [comment]
the new drip is the most important part of this

File: modular_skyrat/modules/colony_fabricator/code/appliances/gps_beacon.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/obj/item/gps/computer/beacon
+	name = "GPS beacon"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/appliances/gps_beacon.dm]:
```suggestion
	name = "\improper GPS beacon"
```

File: modular_skyrat/modules/colony_fabricator/code/clothing/clothing.dm [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+/obj/item/clothing/head/frontier_colonist_helmet
+	name = "frontier soft helmet"
+	desc = "A unusual piece of headwear somewhere between a proper helmet and a normal cap."
+	icon = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing.dmi'
+	icon_state = "tanker"
+	worn_icon = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing_worn.dmi'
+	supports_variations_flags = CLOTHING_SNOUTED_VARIATION_NO_NEW_ICON
+	worn_icon_teshari = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing_worn_teshari.dmi'
+	worn_icon_state = "tanker"
+	inhand_icon_state = null
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/clothing/clothing.dm]: Don't we have default inhand_icon_states especially so they don't just show up as invisible when people hold them in their hands? You could just copy the one that's used almost everywhere, the little bag or whatever, it'd be more consistent with everything else.

File: modular_skyrat/modules/colony_fabricator/code/clothing/mod.dm [hunk 3]
```diff
@@ -0,0 +1,98 @@
+// Its modsuiting time
+
+/datum/mod_theme/frontier_colonist
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/clothing/mod.dm:3]: So this is basically a MODsuit that you can print easily, fully constructed, that can fit in your backpack when you're not using it and that gives you very high resistance to heat, electricity immunity and no slowdown when worn, on top of being twice as fast to retract and deploy? Why would, say, an engineer not want this over a regular engineering MODsuit? This feels like a *big* power-creep, for something that's accessible for the crew nearly every round. I don't see why, as an assistant, I'd bother going to Robotics to get a MODsuit when I could just print this one, and then order a jetpack module, and be ready to go out space exploring.

File: modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	desc = "A rugged backpack often used by settlers and explorers. Holds all of your equipment and then some."
+	icon = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing.dmi'
+	icon_state = "backpack"
+	worn_icon = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing_worn.dmi'
+	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
+	worn_icon_teshari = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing_worn_teshari.dmi'
+	worn_icon_state = "backpack"
+	inhand_icon_state = null
+	// These have the storage of a duffelbag, however do not have the benefits that zipping/unzipping give a duffelbag (like bulky item storage)
+	storage_type = /datum/storage/duffel
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm]: So it's basically just a flatly better backpack? I don't really see how that's balanced, or why everyone wouldn't want to use this. This feels like an easy choice for anyone that wants more storage, over just wearing a duffel bag, as those are still more inconvenient than this.

File: modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+/obj/item/storage/backpack/industrial/frontier_colonist
+	name = "frontier backpack"
+	desc = "A rugged backpack often used by settlers and explorers. Holds all of your equipment and then some."
+	icon = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing.dmi'
+	icon_state = "backpack"
+	worn_icon = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing_worn.dmi'
+	supports_variations_flags = CLOTHING_DIGITIGRADE_VARIATION_NO_NEW_ICON
+	worn_icon_teshari = 'modular_skyrat/modules/colony_fabricator/icons/clothes/clothing_worn_teshari.dmi'
+	worn_icon_state = "backpack"
+	inhand_icon_state = null
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm]: It'd be nice if they had inhand icon states, that just feels like these items are incomplete, compared to every other items in the game trying their hardest to have at least some form of visual representation on the character when inhand.

File: modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	AddElement(/datum/element/manufacturer_examine, COMPANY_FRONTIER)
+	atom_storage.max_slots = 7
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+	// Can hold whatever a toolbelt can + some mining equipment for convenience
+	atom_storage.set_holdable(list(
+		/obj/item/airlock_painter,
+		/obj/item/analyzer,
+		/obj/item/assembly/signaler,
+		/obj/item/clothing/gloves,
+		/obj/item/construction,
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm:52]: This also means that it allows the plumbing equipment to be held in this belt.

File: modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	worn_icon_state = "harness"
+	inhand_icon_state = null
+
+/obj/item/storage/belt/utility/frontier_colonist/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/manufacturer_examine, COMPANY_FRONTIER)
+	atom_storage.max_slots = 7
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+	// Can hold whatever a toolbelt can + some mining equipment for convenience
+	atom_storage.set_holdable(list(
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm:47]: This belt feels overall pretty strong, it's basically just a better toolbelt, unless I'm missing something. Just feels like a big power creep again.

File: modular_skyrat/modules/colony_fabricator/code/design_datums/organic_printer_designs/resources.dm [hunk 8]
```diff
@@ -0,0 +1,10 @@
+/datum/design/organic_plastic
+	name = "Plastic Sheet"
+	id = "oganic_plastic"
+	build_type = BIOGENERATOR
+	materials = list(/datum/material/biomass = 25)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/design_datums/organic_printer_designs/resources.dm:5]: I'm still not entirely sold on this, I don't know, it feels a bit too easy to get plastic. I'll have to think over it some more.

File: modular_skyrat/modules/colony_fabricator/code/design_datums/wearable_equipment.dm [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+/datum/design/frontier_hazard_suit
+	name = "Frontier Hazard Protective MOD Control Unit"
+	desc = "The pinnacle of frontier cheap technology. Suits such as these are made specifically for the rare emergency that creates a hazard \
+		environment that other equipment just can't quite handle. Often, these suits are able to protect their users \
+		from not only electricity, but also radiation, biological hazards, other people, so on. This suit will not, \
+		however, protect you from yourself."
+	id = "frontier_hazard_suit"
+	build_type = COLONY_FABRICATOR
+	// This materials list is a slightly cheaper compilation of this modsuit's constituent parts
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/code/design_datums/wearable_equipment.dm]: How much cheaper are we talking? A few sheets? Or like half a sheet of this, a quarter sheet of that, etc.?

File: modular_skyrat/modules/company_imports/code/armament_datums/akh_frontier.dm [hunk 10]
```diff
@@ -21,6 +21,28 @@
...
+	subcategory = "Wearable Equipment"
+
+/datum/armament_entry/company_import/akh_frontier/wearables/hazard_mod
+	name = "Frontier Hazard Protective MOD Control Unit"
+	description = "The pinnacle of frontier cheap technology. Suits such as these are made specifically for the rare emergency that creates a hazard \
+		environment that other equipment just can't quite handle. Often, these suits are able to protect their users \
+		from not only electricity, but also radiation, biological hazards, other people, so on. This suit will not, \
+		however, protect you from yourself."
+	item_type = /obj/item/mod/control/pre_equipped/frontier_colonist
+	cost = PAYCHECK_COMMAND * 6.5
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/company_imports/code/armament_datums/akh_frontier.dm:36]: This is ridiculously cheap compared to all of the other available MODsuit options, for reference the Engineering MODsuit costs about 2600 in the regular Cargo console orders.

File: modular_skyrat/modules/colony_fabricator/sound/attributions.txt [hunk 11]
```diff
@@ -12,3 +12,7 @@ modular_skyrat/modules/colony_fabricator/sound/conditioner/conditioner_(1 - 4).w
 The following sounds are from https://pixabay.com/sound-effects/welder-3-54547/
 
 modular_skyrat/modules/colony_fabricator/sound/arc_welder/arc_welder.ogg
+
+The following sounds are from https://pixabay.com/sound-effects/sos-signal-137144/
+
+modular_skyrat/modules/colony_fabricator/sound/morse_signal.wav
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/colony_fabricator/sound/attributions.txt:18]: I haven't see it, where does this sound actually come into play in all of this?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I feel like there's a lot of powercreep to this. While a lot of this could be more than acceptable for events, the fact that it's all freely available to the crew just makes it the go-to for a lot of things, as they're just either stupidly easy to get as opposed to every other method of acquiring them (or a lot cheaper than the other alternatives), or they're just flat-out upgrades over what the crew normally has access to.

File: modular_skyrat/modules/colony_fabricator/sound/attributions.txt (hunk 11 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/colony_fabricator/sound/attributions.txt:18]: This is the headset's radio sound effect, the one it makes when you speak over comms

File: modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm (hunk 4 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm]: Not sure why I did that one, its just normal backpack storage now.

File: modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm (hunk 6 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/colony_fabricator/code/clothing/storage.dm:52]: Considering there's engineering plumbing equipment and I don't think being able to hold plumbing stuff is like, _strong_ or anything, I think its fine.

File: modular_skyrat/modules/colony_fabricator/code/design_datums/organic_printer_designs/resources.dm (hunk 8 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/colony_fabricator/code/design_datums/organic_printer_designs/resources.dm:5]: Consider that the main source of plastic is someone going to a chem machine, pressing like five buttons and having as much plastic as they want for practically free forever. This is frankly worse than using the intended way to get it (outside of the fact cargo can order 50 sheets of plastic for like 800 credits I think)

File: modular_skyrat/modules/colony_fabricator/code/design_datums/wearable_equipment.dm (hunk 9 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/colony_fabricator/code/design_datums/wearable_equipment.dm]: I've changed it to be the exact cost of everything its made of, no cheaper except that the cost of the tiny cell in it isnt taken into account

File: modular_skyrat/modules/company_imports/code/armament_datums/akh_frontier.dm (hunk 10 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/armament_datums/akh_frontier.dm:36]: There are enough downsides piled onto this suit that I think anyone doing stuff that you would want, say, an engineering modsuit for would rather have the engi mod over this regardless of how cheap this is.

File: modular_skyrat/modules/colony_fabricator/code/clothing/mod.dm (hunk 3 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/colony_fabricator/code/clothing/mod.dm:3]: I've made some changes to it, I'll tell you over discord

**Pull request closed without merging**